apps/davi/src/Modules/Guilds/pages/EditDiscussion.tsx

```diff
@@ -0,0 +1,224 @@
+import { useOrbisContext } from 'contexts/Guilds/orbis';
+import { IconButton } from 'components/primitives/Button';
+import { StyledLink } from 'components/primitives/Links';
+import { linkStyles } from './Proposal/Proposal.styled';
+import { useTheme } from 'styled-components';
+import { ProposalDescription } from 'components/ProposalDescription';
+import { WalletModal } from 'components/Web3Modals';
+import { useAccount } from 'wagmi';
+import { isReadOnly } from 'provider/wallets';
+
+const EditDiscussionPage: React.FC = () => {
+  const { orbis } = useOrbisContext();
+
+  const { chainName: chain, guildId, discussionId } = useTypedParams();
+  const { isLoading: isGuildAvailabilityLoading } = useContext(
+    GuildAvailabilityContext
+  );
+
+  const navigate = useNavigate();
+  const { t } = useTranslation();
+  const [user, setUser] = useState('');
+  const [editMode, setEditMode] = useState(true);
+  const [title, setTitle] = useState('');
+  const [streamId, setStreamId] = useState<string>('');
+  const [html, onHTMLChange] = useState<string>();
+  const [initialDescription, setInitialDescription] = useState<string>('');
+  const [isWalletModalOpen, setIsWalletModalOpen] = useState(false);
+
+  const theme = useTheme();
+
+  const {
+    connector,
+    isConnecting: isWalletConnecting,
+    isConnected: isWalletConnected,
+  } = useAccount();
+
+  useEffect(() => {
+    isConnected(orbis).then(res => {
+      if (res) {
+        console.log('Already connected with: ', res);
+      } else {
+        connect(orbis).then(did => {
+          setUser(did);
+        });
+      }
+    });
+  }, [user, orbis]);
+
+  const getPost = async () => {
+    const { data } = await orbis.getPost(discussionId);
+    setTitle(data?.content?.title);
+    setInitialDescription(data?.content?.body);
+    setStreamId(data?.stream_id);
+  };
+
+  useEffect(() => {
+    getPost();
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  const {
+    Editor,
+    EditorConfig,
+    md: discussionBodyMd,
+    html: discussionBodyHtml,
+  } = useTextEditor(
+    t('discussions.discussionPlaceholder'),
+    onHTMLChange,
+    html,
+    initialDescription
+  );
+
+  const hasWalletConnection = useMemo(() => {
+    return !isWalletConnecting && !isReadOnly(connector) && isWalletConnected;
+  }, [connector, isWalletConnected, isWalletConnecting]);
+
+  const toggleWalletModal = () => {
+    setIsWalletModalOpen(!isWalletModalOpen);
+  };
+
+  const handleToggleEditMode = () => {
+    if (editMode && !title.trim() && !discussionBodyMd.trim()) return;
+    setEditMode(v => !v);
+  };
+
+  const handleBack = () =>
+    navigate(`/${chain}/${guildId}/discussion/${discussionId}`);
+
+  const handleEditDiscussion = async (post: DiscussionContent) => {
+    if (!hasWalletConnection) {
+      setIsWalletModalOpen(true);
+      return '';
+    }
+
+    if (postTemplate(post)) {
+      const res = await editPost(orbis, streamId, post);

```

Comment:
Is `streamId` and `discussionId` the same? In that case, maybe stick to one of them for coherence. If not, you can ignore this comment :).

Reply:
Yeah, these are the same. Will stick to one.